docs/mkdocs.yml

```diff
@@ -38,6 +38,7 @@ nav:
   - API:
     - Quickstart: java-api-quickstart.md
     - API: api.md
+    - File I/O: ../../fileio/
```

Comment:
weird thing i noticed while rendering this locally. 

api.md / https://iceberg.apache.org/docs/nightly/api/
has a side bar

but fileio does not. 


i didnt see where this is controlled but would be good to keep the consistency here